prow/test/integration/test/webhook_server_test.go

```diff
@@ -123,3 +127,66 @@ func TestWebhookServerMutateProwjob(t *testing.T) {
 		t.Errorf("prowjob was not mutated %v", err)
 	}
 }
+
+func TestWebhookServerAvailability(t *testing.T) {
+	clusterContext := getClusterContext()
+	kubeClient, err := NewClients("", clusterContext)
+	if err != nil {
+		t.Fatalf("Failed creating clients for cluster %q: %v", clusterContext, err)
+	}
+	ctx := context.Background()
+	var podList corev1.PodList
+	webhookServerPod := &corev1.Pod{}
+	err = kubeClient.List(ctx, &podList, &client.ListOptions{})
+	if err != nil {
+		t.Fatalf("could not list pods %v", err)
+	}
+	for _, pod := range podList.Items {
+		if pod.ObjectMeta.Labels["app"] == "webhook-server" {
+			webhookServerPod = &pod
+			break
+		}
+	}
+	count := 0
+	for start := time.Now(); time.Since(start) < time.Second * 30; {
+		absProwJobPath, err := filepath.Abs(validBareProwJobPath)
+		if err != nil {
+			t.Fatalf("could not find absolute file path %v", err)
+		}
+		//read yaml file
+		prowJobYAML, err := ioutil.ReadFile(absProwJobPath)
+		if err != nil {
+			t.Fatalf("unable to read yaml file %v", err)
+		}
+		var prowJob v1.ProwJob
+		err = yaml.Unmarshal(prowJobYAML, &prowJob)
+		if err != nil {
+			t.Fatalf("unable to unmarshal into prowjob %v", err)
+		}
+		//update foo label of prowjob
+		val := prowJob.Labels["foo"]
+		prowJob.Labels["foo"] = val + strconv.Itoa(count)
+		newProwJobYAML, err := yaml.Marshal(&prowJob)
+		if err != nil {
+			t.Fatalf("could not marshal prowjob %v", err)
+		}
+		//write to new yaml to file
+		if err := ioutil.WriteFile("./testdata/updated_prowjob.yaml", newProwJobYAML, 0666); err != nil {
+			t.Fatalf("could not write contexts of file")
+		}
+		//apply prowjob to cluster
+		out, err := exec.Command("kubectl", "--context="+getClusterContext(), "apply", "-f", "./testdata/updated_prowjob.yaml").CombinedOutput()
+		if err != nil {
+			t.Fatalf("prowjob was not properly validated. Error: %v Output: %s", err, string(out))
+		}
+		count++;
+		go func() {
+			out, err := exec.Command("kubectl", "--context="+getClusterContext(), "delete", "pod", webhookServerPod.GetGenerateName()).CombinedOutput()
+			if err != nil {
+				t.Fatalf("could not delete pod from cluster. Error: %v Output: %s", err, string(out))
+			}
+			time.Sleep(time.Second)
```

Comment:
We should sleep before the deletion, not after. This doesn't really have an effect because the goroutine ends immediately after sleeping.